wandb/sdk/artifacts/storage_handlers/s3_handler.py

```diff
@@ -134,9 +138,18 @@ def load_path(
                     )
                 )
 
-        with cache_open(mode="wb") as f:
-            obj.download_fileobj(f, ExtraArgs=extra_args)
-        return path
+        size_of_file_gb = manifest_entry.size / 1024 / 1024 / 1024
+        max_concurrency = 10 + max(size_of_file_gb // 1, 20)
+        config = self._boto_transfer.TransferConfig(max_concurrency=max_concurrency)
+        if manifest_entry.skip_cache_copy:
+            dest_path = os.path.join(manifest_entry._parent_artifact._default_root(), manifest_entry.path)
+            with open(dest_path, 'wb') as f:
+                obj.download_fileobj(f, ExtraArgs=extra_args, Config=config)
```

Comment:
so this config only affects one s3 client right? I know we rely on asyncio and parallelism for downloads so I'm curious on the effect of this for many small files. It might be worthwhile to do a test of this branch with lots of small files to make sure we dont have regressions

Reply:
If it were a small file the max_concurrency should stay at 10 (which is our current boto s3 default) though i agree these params we should play with more.